hack/tools/tilt-prepare/main.go

```diff
@@ -775,6 +775,10 @@ func prepareWorkload(name, prefix, binaryName, containerName string, objs []unst
 			cmd := []string{"sh", "/start.sh", "/" + binaryName}
 			args := append(container.Args, []string(ts.ExtraArgs[name])...)
 
+			// remove securityContext for tilt live_update, see https://github.com/tilt-dev/tilt/issues/3060
+			d.Spec.Template.Spec.SecurityContext = nil
```

Comment:
Yes I wasn't sure because if I set it out of the loop, it means removing `.spec.securityContext` for every deployments returned by kustomize without taking in account the `containerName` arg. (`updateDeployment` iterates on all objects and execute this function)

This means that all deployments won't have `.spec.template.spec.securityContext` but may have `spec.template.spec.containers[].securityContext`.
This shouldn't be an issue per say but may be misleading if a provider adds more deployments.

As is only container matching `containerName` will have the securityContext, and it will be ensure on its parent template spec.

I can add a comment to explain why it's modified in the loop or make the change globally on all deployments (+every containers maybe so it's not misleading), let me know

Reply:
Oh I missed that. Thx for the explanation :)

Maybe a short comment that this is only intentionally done for Deployments that have a manager would be good

Reply:
Just pushed a new revision, let me know if that comment is clear enough :wink:

Reply:
Perfect, thx!

Reply:
Mostly just a safeguard if someone else comes along in the future and wants to "optimize" it :)